float32[4] joint
string gripper